methods.py

```diff
@@ -130,9 +130,10 @@ def disable_warnings(self):
     if self.msvc and not using_clang(self):
         # We have to remove existing warning level defines before appending /w,
         # otherwise we get: "warning D9025 : overriding '/W3' with '/w'"
-        self["CCFLAGS"] = [x for x in self["CCFLAGS"] if not (x.startswith("/W") or x.startswith("/w"))]
-        self["CFLAGS"] = [x for x in self["CFLAGS"] if not (x.startswith("/W") or x.startswith("/w"))]
-        self["CXXFLAGS"] = [x for x in self["CXXFLAGS"] if not (x.startswith("/W") or x.startswith("/w"))]
+        WARN_FLAGS = ["/Wall", "/W4", "/W3", "/W2", "/W1", "/W0"]
+        self["CCFLAGS"] = [x for x in self["CCFLAGS"] if x not in WARN_FLAGS]
```

Comment:
Won't this lead to warnings if you have all the `disabled_warnings` from SCons + `/w`?
MSVC seems to be really finicky about having any kind of conflicting or double specification of expected behavior.

Reply:
Surprisingly, no! The conflict warnings only crop up for the main overrides; individual warnings & even warn-as-error can still be passed without issue